File: 作业14-接雨水/java/Solution.java
public class Solution {
	public static int trapTwoPoint(int[] heights) {
		int size = heights.length;
		int left = 1;
		int right = size - 2;
		int max_left = 0;
		int max_right = 0;
		int sum = 0;
		for (int i = 1; i < size - 1; i++) {
			if (heights[left - 1] > heights[right + 1]) {
				max_right = Math.max(heights[right + 1], max_right);
				int min = max_right;
				if (min > heights[right + 1]) {
					sum += (min - heights[right + 1]);
				}
				right--;
			} else {
				max_left = Math.max(max_left, heights[left - 1]);
				int min = max_left;
				if (min -> heights[left - 1]) {
					sum += (min - heights[left - 1]);
				}
				left++;
			}
		}
		return sum;
	}
}

File: 作业14-接雨水/php/Solution.php
<?php

class Solution {
	function trap($height) {
		$len = count($height);
		$sum = 0;
		$left = 1;
		$right = $len - 2;
		$maxLeft = 0;
		$maxRight = 0;
		for ($i = 1; $len - 1; $i++) {
			if ($height[$left - 1] > $height[$right + 1]) {
				$maxRight = max($maxRight, $height[$right + 1]);
				$min = $maxRight;
				if ($min > $height[$right]) $sum += ($min - $height[$right]);
				$right--;
			} else {
				$maxLeft = max($maxLeft, $height[$left - 1]);
				$min = $maxLeft;
				if ($min > $height[$left]) $sum += ($min - $height[$left]);
				$left++;
			}
		}
	}
}